app/src/main/java/com/larryhsiao/nyx/android/jot/NewJotFragment.java

```diff
@@ -0,0 +1,162 @@
+package com.larryhsiao.nyx.android.jot;
+
+import android.app.AlertDialog;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.*;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.RecyclerView;
+import com.google.android.material.chip.Chip;
+import com.google.android.material.chip.ChipGroup;
+import com.larryhsiao.nyx.BuildConfig;
+import com.larryhsiao.nyx.R;
+import com.larryhsiao.nyx.android.base.JotFragment;
+import com.larryhsiao.nyx.attachments.NewAttachments;
+import com.larryhsiao.nyx.jots.Jot;
+import com.larryhsiao.nyx.jots.JotUri;
+import com.larryhsiao.nyx.jots.NewJot;
+import com.larryhsiao.nyx.tags.NewJotTag;
+import com.larryhsiao.nyx.tags.NewTag;
+import com.larryhsiao.nyx.tags.Tag;
+import com.schibstedspain.leku.LocationPickerActivity;
+import com.silverhetch.clotho.source.ConstSource;
+
+import java.util.stream.Collectors;
+
+import static android.app.Activity.RESULT_OK;
+import static android.content.Intent.ACTION_OPEN_DOCUMENT;
+import static com.schibstedspain.leku.LocationPickerActivityKt.*;
+import static java.lang.Double.MIN_VALUE;
+
+/**
+ * Fragment to create new Jot.
+ * <p>
+ * Use this fragment with
+ */
+public class NewJotFragment extends JotFragment {
+    private static final int REQUEST_CODE_LOCATION_PICKER = 1000;
+    private static final int REQUEST_CODE_FILE_PICKER = 1001;
+    private ChipGroup tagGroup;
+    private TextView locationText;
+    private ImageView attachmentIcon;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'attachmentIcon' could be replaced by a local variable.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5029215)